content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md

```diff
@@ -0,0 +1,157 @@
+
+## Pom file configuration
+
+`camel-debug` must be on the classpath. One way to achieve that is to provide the dependency in a profile which needs to be activated when the tests are launched.
+
+```xml
+<profiles>
+    <profile>
+        <id>camel.debug</id>
+        <activation>
+            <property>
+                <name>camel.debug</name>
+                <value>true</value>
+            </property>
+        </activation>
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.camel</groupId>
+                <artifactId>camel-debug</artifactId>
+            </dependency>
+        </dependencies>
+    </profile>
+</profiles>
+```
+
+## Test class configuration
+
+Two methods must have a specific value:
+
+- `isUseDebugger` method must return false. This is the default implementation, just ensure that it is not overridden.
+- `useJMX` method must return true. This needs to be overridden.
+
+
+We also need to ensure that the test is starting after Debugger is attached and breakpoint set. For now, the only solution that I can propose is to insert a (ugly) `Thread.sleep` at the beginning of the test. It will work only if the test is triggering the route execution. If your route is automatically triggered like for a `timer`, some route executions might happen before the debugger is attached and breakpoints enabled.
+
+You should end up with something like that:
+
+```java
+class MainTest extends CamelMainTestSupport {
+	@Test
+	void myTest() throws Exception {
+		Thread.sleep(2000); // to let time to Debugger to attach and install breakpoints
+		template.asyncSendBody("direct:demo", ""); // Take care to start the route in an async way
+		NotifyBuilder notify = new NotifyBuilder(context)
+			.whenCompleted(1).whenBodiesDone("Bye World").create();
+		assertTrue(
+			notify.matches(60, TimeUnit.SECONDS), "1 message should be completed"
+		);
+	}
+
+	@Override
+	protected boolean useJmx() {
+		return true;
+	}
+}
+```
+
+You might also want to introduce a longer timeout to let you time to debug.
+
+# IDE configuration
+
+## VS Code Task configuration
+
+The following VS Code task must be created in the `settings/tasks.json` file:
+
+```json
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"label": "Start test with camel.debug profile",
+			"type": "shell",
+			"command": "mvn", // mvn binary of Maven must be available on command-line
+			"args": [
+				"test",
+				"-Pcamel.debug" // This depends on your project. The goal here is to have camel-debug on the classpath.
+			],
+			"problemMatcher": "$camel.debug.problemMatcher",
+			"presentation": {
+				"reveal": "always"
+			},
+			"isBackground": true // Must be set as background as the Maven commands doesn't return until the Camel application stops.
+		}
+	]
+}
+```
+
+It allows to start the test with `camel-debug` on the classpath.
+
+## VSCode Launch configuration
+
+The following launch configuration must be created in the `.settings/launch.json`:
```

Comment:
@apupier I'm wondering if the right path is rather `.vscode/launch.json`?

Reply:
like in your fork https://github.com/apupier/camel-examples/tree/Blogpost-Example-Debug-Camel-In-Test/examples/main/.vscode

Reply:
created https://github.com/apache/camel-website/pull/863